From dcce62dd525cf0f8e572e56a8990aea7ec2f0ade Mon Sep 17 00:00:00 2001
From: Jeffery To <jeffery.to@gmail.com>
Date: Tue, 30 May 2023 23:47:59 +0800
Subject: [PATCH] unix-ffi/os-path: Add unix-ffi version of `os-path` package.

This package reuses the code from the python-stdlib version of `os-path`
but requires the unix-ffi version of `os`.

This also updates `glob` to require this version of `os-path`.

Signed-off-by: Jeffery To <jeffery.to@gmail.com>
---
 unix-ffi/glob/manifest.py    | 2 +-
 unix-ffi/os-path/manifest.py | 6 ++++++
 unix-ffi/os/os/__init__.py   | 6 ++++++
 3 files changed, 13 insertions(+), 1 deletion(-)
 create mode 100644 unix-ffi/os-path/manifest.py

--- a/unix-ffi/glob/manifest.py
+++ b/unix-ffi/glob/manifest.py
@@ -1,7 +1,7 @@
 metadata(version="0.5.2")
 
 require("os", unix_ffi=True)
-require("os-path")
+require("os-path", unix_ffi=True)
 require("re", unix_ffi=True)
 require("fnmatch")
 
--- /dev/null
+++ b/unix-ffi/os-path/manifest.py
@@ -0,0 +1,6 @@
+metadata(version="0.1.4")
+
+# Originally written by Paul Sokolovsky.
+
+require("os", unix_ffi=True)
+package("os", base_path="../../python-stdlib/os-path")
--- a/unix-ffi/os/os/__init__.py
+++ b/unix-ffi/os/os/__init__.py
@@ -5,6 +5,12 @@ import stat as stat_
 import ffilib
 import uos
 
+# Provide optional dependencies (which may be installed separately).
+try:
+    from . import path
+except ImportError:
+    pass
+
 R_OK = const(4)
 W_OK = const(2)
 X_OK = const(1)
